pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -4959,84 +4963,120 @@ spec:
 		{
 			ObjectMeta: metav1.ObjectMeta{Name: config.GetFeatureFlagsConfigName(), Namespace: system.Namespace()},
 			Data: map[string]string{
 				"trusted-resources-verification-no-match-policy": config.FailNoMatchPolicy,
+				"enable-tekton-oci-bundles":                      "true",
 			},
 		},
 	}
 
 	testCases := []struct {
-		name          string
-		taskrun       []*v1beta1.TaskRun
-		task          []*v1beta1.Task
-		expectedError error
+		name      string
+		taskBytes []byte
 	}{
 		{
-			name:          "unsigned task fails verification",
-			task:          []*v1beta1.Task{ts},
-			expectedError: trustedresources.ErrResourceVerificationFailed,
+			name:      "unsigned task fails verification",
+			taskBytes: unsignedTaskBytes,
 		},
 		{
-			name:          "modified task fails verification",
-			task:          []*v1beta1.Task{tamperedTask},
-			expectedError: trustedresources.ErrResourceVerificationFailed,
+			name:      "modified task fails verification",
+			taskBytes: modifiedTaskBytes,
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
+			tr := parse.MustParseV1beta1TaskRun(t, fmt.Sprintf(`
+metadata:
+  name: test-taskrun
+  namespace: foo
+spec:
+  params:
+  - name: myarg
+    value: foo
+  taskRef:
+    resolver: %s
+    name: test-task
+status:
+  podName: the-pod
+`, resolverName))
+
 			d := test.Data{
 				TaskRuns:             []*v1beta1.TaskRun{tr},
-				Tasks:                tc.task,
 				ConfigMaps:           cms,
 				VerificationPolicies: vps,
 			}
 
 			testAssets, cancel := getTaskRunController(t, d)
 			defer cancel()
 			createServiceAccount(t, testAssets, tr.Spec.ServiceAccountName, tr.Namespace)
-			err := testAssets.Controller.Reconciler.Reconcile(testAssets.Ctx, getRunName(tr))
+			err = testAssets.Controller.Reconciler.Reconcile(testAssets.Ctx, getRunName(tr))
 
-			if !errors.Is(err, tc.expectedError) {
-				t.Errorf("Reconcile got %v but want %v", err, tc.expectedError)
-			}
-			tr, err := testAssets.Clients.Pipeline.TektonV1beta1().TaskRuns(tr.Namespace).Get(testAssets.Ctx, tr.Name, metav1.GetOptions{})
-			if err != nil {
-				t.Fatalf("getting updated taskrun: %v", err)
-			}
-			condition := tr.Status.GetCondition(apis.ConditionSucceeded)
+			reconciledRun := resolveResolutionRequestandReconcile(t, testAssets, resolverName, tc.taskBytes, tr, true)
+			condition := reconciledRun.Status.GetCondition(apis.ConditionSucceeded)
 			if condition.Type != apis.ConditionSucceeded || condition.Status != corev1.ConditionFalse || condition.Reason != podconvert.ReasonResourceVerificationFailed {
 				t.Errorf("Expected TaskRun to fail with reason \"%s\" but it did not. Final conditions were:\n%#v", podconvert.ReasonResourceVerificationFailed, tr.Status.Conditions)
 			}
 		})
 	}
 }
+
+func resolveResolutionRequestandReconcile(t *testing.T, testAssets test.Assets, resolverName string, resourceBytes []byte, tr *v1beta1.TaskRun, permanentError bool) *v1beta1.TaskRun {
```

Comment:
It looks like this is supposed to be a mock resolver, is that right? Can this be made more generic, or are there any existing mock resolvers that could work here?

Reply:
Yes this is a mock resolver. I'm reusing the same code snippet from https://github.com/tektoncd/pipeline/blob/f525f721191f9debb29daf2cb23aa42dc109ba7b/pkg/reconciler/pipelinerun/pipelinerun_test.go#L6743

Reply:
I took a look at the existing fake resolver (https://github.com/tektoncd/pipeline/blob/main/pkg/resolution/resolver/framework/fakeresolver.go) and I don't think it's quite what we want here, but this is a complex helper function to create just for a few tests.

I think it would be cleaner to seed the test data with an already resolved resolution request containing the data we need. Can you see if you can make that approach work?

If that's really not possible, can you make this helper function more generic, so that it works for other tests as well, create an issue for a mock resolver, and link it as a comment here and in TestReconcileWithResolver?

Reply:
Oh sorry, I think we are "seeding the test data with an already resolved resolution request"? There's a "foo" resolver there so that's why I said it was mock resolver (this is not the fake resolver), but it is more accurate to say that we are using mock resolution request here.

Reply:
What I meant was passing the resolution requests into the `test.Data` struct and testing only a single reconcile loop; would that work?

Reply:
I just tested to use `test.Data` and it should work, I'm updating all the tests now

Reply:
Updated